examples/skybox/main.rs

```diff
@@ -0,0 +1,303 @@
+            }),
+            primitive_topology: wgpu::PrimitiveTopology::TriangleList,
+            color_states: &[wgpu::ColorStateDescriptor {
+                format: sc_desc.format,
+                color_blend: wgpu::BlendDescriptor::REPLACE,
+                alpha_blend: wgpu::BlendDescriptor::REPLACE,
+                write_mask: wgpu::ColorWrite::ALL,
+            }],
+            depth_stencil_state: None,
+            index_format: wgpu::IndexFormat::Uint16,
+            vertex_buffers: &[wgpu::VertexBufferDescriptor {
+                stride,
+                step_mode: wgpu::InputStepMode::Vertex,
+                attributes: &attributes,
+            }],
+            sample_count: 1,
+            sample_mask: !0,
+            alpha_to_coverage_enabled: false,
+        });
+
+        let sampler = device.create_sampler(&wgpu::SamplerDescriptor {
+            address_mode_u: wgpu::AddressMode::ClampToEdge,
+            address_mode_v: wgpu::AddressMode::ClampToEdge,
+            address_mode_w: wgpu::AddressMode::ClampToEdge,
+            mag_filter: wgpu::FilterMode::Nearest,
+            min_filter: wgpu::FilterMode::Linear,
+            mipmap_filter: wgpu::FilterMode::Nearest,
+            lod_min_clamp: -100.0,
+            lod_max_clamp: 100.0,
+            compare_function: wgpu::CompareFunction::Always,
+        });
+
+        let paths = [
+            "examples/skybox/images/posx.png",
+            "examples/skybox/images/posy.png",
+            "examples/skybox/images/posz.png",
+            "examples/skybox/images/negx.png",
+            "examples/skybox/images/negy.png",
+            "examples/skybox/images/negz.png",
+        ];
+	// we set these multiple times, but whatever
+	let (mut image_width, mut image_height) = (0, 0);
+        let faces = paths
+            .iter()
+            .map(|path| {
+                let f = std::fs::File::open(path).expect("can open file");
+                let decoder = png::Decoder::new(f);
+                let (info, mut reader) = decoder.read_info().expect("can read info");
+                image_width = info.width;
+                image_height = info.height;
+                let mut buf = vec![0; info.buffer_size()];
+                reader.next_frame(&mut buf).expect("can read png frame");
+                buf
+            })
+            .collect::<Vec<_>>();
+
+        let texture_extent = wgpu::Extent3d {
+            width: image_width,
+            height: image_height,
+            depth: 1,
+        };
+
+        let texture = device.create_texture(&wgpu::TextureDescriptor {
+            size: texture_extent,
+            array_layer_count: 6,
+            mip_level_count: 1,
+            sample_count: 1,
+            dimension: wgpu::TextureDimension::D2,
+            format: SKYBOX_FORMAT,
+            usage: wgpu::TextureUsage::SAMPLED
+                | wgpu::TextureUsage::COPY_DST,
+        });
+
+        {
+            let mut encoder =
```

Comment:
This encoder is dropped without being submitted. Can we just use `init_encoder` instead?

Reply:
good catch; so i just fixed this, and still not showing anything

Reply:
the texture now shoes up in gpu profiling though, so that's a good sign